### PR TITLE
Fix completion issues within the object field context

### LINKED ISSUE #31618: Object field context completions are not accurate.
**Description:**
$title
1. Need to fix the completions for qualified name context
2. Need to implement the context aware sorting

**Steps to reproduce:**

**Affected Versions:**
swanlake beta2 at least

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #31618

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/CheckExpressionNodeContext.java [hunk 1]
```diff
@@ -48,19 +44,15 @@ public CheckExpressionNodeContext() {
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext ctx, CheckExpressionNode node)
             throws LSCompletionException {
         List<LSCompletionItem> completionItems = new ArrayList<>();
-        NonTerminalNode nodeAtCursor = ctx.getNodeAtCursor();
         if (node.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT
                 || node.parent().kind() == SyntaxKind.LOCAL_VAR_DECL
-                || node.parent().kind() == SyntaxKind.MODULE_VAR_DECL) {
+                || node.parent().kind() == SyntaxKind.MODULE_VAR_DECL
+                || node.parent().kind() == SyntaxKind.OBJECT_FIELD) {
```
Comment (pubudu91) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/CheckExpressionNodeContext.java:50]: Not related to this PR, but just curious, why only object fields? What about record fields? 🤔

**Review by pubudu91 — APPROVED** [review]

**Comment by TharushiJay** [comment]
LGTM!

**Comment by gayalkuruppu** [comment]
LGTM

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/CheckExpressionNodeContext.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/CheckExpressionNodeContext.java:50]: That is coming with a separate PR :)

**Pull request merged**